File: Binary tree/Depth_first_Search/javascript/InOrderTraversal.js
class Node {
  constructor(data) {
    this.data = data;
    this.left = null;
    this.right = null;
  }
}
class InorderTraversal {
  Inorder(root) {
    if (root === null) {
      return;
    }

    this.Inorder(root.left);
    console.log(root.data);
    this.Inorder(root.right);
  }
}

function main() {
  let root = new Node(1);
  root.left = new Node(2);
  root.right = new Node(3);
  root.left.left = new Node(4);
  root.left.right = new Node(5);
  root.right.left = new Node(6);
  root.right.right = new Node(7);

  let traversal = new InorderTraversal();
  traversal.Inorder(root);
}

main();

File: Binary tree/Depth_first_Search/java/InOrderTraversal.java
class Node{
    int data;
    Node left;
    Node right;

    public Node(int data){
        this.data= data;
        left = null;
        right =null;
    }
}

public class InOrderTraversal{

    public void InOrder(Node root){
        if(root == null){
            return;
        }
        InOrder(root.left);
        System.out.println(root.data);
        InOrder(root.right);
    }


    public static void main(String args[]){
        Node root =new Node(1);
        root.left= new Node(2);
        root.right = new Node(3);
        root.left.left = new Node(4);
        root.left.right= new Node(5);
        root.right.left=new Node(6);
        root.right.right=new Node(7);

        InOrderTraversal traversal = new InOrderTraversal();
        traversal.InOrder(root);


    }
}